### PR TITLE
[core] subprocess debug log messages

### LINKED ISSUE #1286: Add more debug logging for base checks
I'm trying to debug an issue where an Agent is self-destructing, but even with DEBUG logging on you aren't able to tell what system checks are actually running...here's an example (hostnames removed for security):

```
# Some init logging...
...
2014-11-26 06:48:04 UTC | INFO | dd.collector | config(config.py:824) | initialized checks.d checks: ['network']
2014-11-26 06:48:04 UTC | INFO | dd.collector | config(config.py:825) | initialization failed checks.d checks: []
2014-11-26 06:48:04 UTC | INFO | dd.collector | checks.collector(datadog.py:67) | Dogstream parsers: []
2014-11-26 06:48:04 UTC | DEBUG | dd.collector | util(util.py:437) | Resetting watchdog for 150
2014-11-26 06:48:04 UTC | DEBUG | dd.collector | checks.collector(collector.py:133) | Starting collection run #1
2014-11-26 06:48:04 UTC | DEBUG | dd.collector | config(config.py:571) | No proxy configured
2014-11-26 06:48:04 UTC | INFO | dd.collector | checks.collector(collector.py:477) | Hostnames: {'socket-hostname': '******', 'timezones': ('UTC', 'UTC'), 'hostname': '****', 'socket-fqdn': '****'}, tags: {'google cloud platform': [****]}
2014-11-26 06:50:34 UTC | ERROR | dd.collector | util(util.py:424) | Self-destructing...
2014-11-26 06:50:34 UTC | ERROR | dd.collector | util(util.py:425) | None
```

I'd love to see:

1. A log line for when a check starts and stops.
2. Maybe some logging inside of the system checks or even logging of shell commands that are being run?

### PR DESCRIPTION
**Pull request by yannmh**
Adding more logging inside system checks to help future investigations.
Fix #1286.

### CODE REVIEW

File: util.py [hunk 1]
```diff
@@ -43,6 +44,24 @@
 NumericTypes = (float, int, long)
 
 
+def log_subprocess(func):
```
Comment (degemer) [inline util.py]: Could you put this function and `import subprocess` in `utils/` ?
I think we are slowly deprecating `util.py` in favor of `utils/`, more precise/beauteous.

Comment (yannmh) [inline util.py]: Moved to `utils.subprocess_output`. Thanks !

File: utils/subprocess_output.py [hunk 2]
```diff
@@ -21,3 +25,21 @@ def get_subprocess_output(command, log):
...
+    """
+    @wraps(func)
+    def wrapper(*params, **kwargs):
+        fc = "%s(%s)" % (func.__name__, ', '.join(
+            [a.__repr__() for a in params] +
+            ["%s = %s" % (a, repr(b)) for a, b in kwargs.items()]
+        ))
+        log.debug("%s called" % fc)
+        result = func(*params, **kwargs)
+        return result
```
Comment (LeoCavaille) [inline utils/subprocess_output.py]: nitpick but `return func()`

File: utils/subprocess_output.py [hunk 3]
```diff
@@ -21,3 +25,21 @@ def get_subprocess_output(command, log):
...
+
+def log_subprocess(func):
+    """
+    Wrapper around subprocess to log.debug commands.
+    """
+    @wraps(func)
+    def wrapper(*params, **kwargs):
+        fc = "%s(%s)" % (func.__name__, ', '.join(
+            [a.__repr__() for a in params] +
+            ["%s = %s" % (a, repr(b)) for a, b in kwargs.items()]
```
Comment (LeoCavaille) [inline utils/subprocess_output.py]: nitpick: `%r` calls `repr()` for you

**Comment by yannmh** [comment]
Thanks for your feedback. Merging

**Comment by conorbranagan** [comment]
:+1: Thanks for fixing this one!

**Pull request merged**